### ISSUE TITLE
bug: 2.4.0 `Cargo.lock` is inconsistent with its `Cargo.toml`

### ISSUE THREAD

**Issue by RaitoBezarius** [starkware-libs/cairo#4530]
# Bug Report

**Cairo version:**

2.4.0.

**Current behavior:**

It's impossible to perform `cargo build --frozen`.

**Expected behavior:**

Given all the dependencies described in the Cargo.lock, `cargo build --frozen` should succeed.

**Steps to reproduce:**

The perfect reproducer is to try the Nixpkgs packaging of Cairo compiler and bump it to 2.4.0 naively, it will fail.

**Related code:**

Here's a fragment of the error trace:

```
cairo> building '/nix/store/p6652d5bzpzgkqzvqvx3xfbijh8c1n3a-cairo-2.4.0.drv' on 'ssh://build01-x86-newtype'
copying 0 paths
cairo> unpacking sources
cairo> unpacking source archive /nix/store/g1dmsn88z50rqajlbldlngh1nj0xf7zi-source
cairo> source root is source
cairo> Executing cargoSetupPostUnpackHook
cairo> Finished cargoSetupPostUnpackHook
cairo> patching sources
cairo> Executing cargoSetupPostPatchHook
cairo> Validating consistency between /build/source/Cargo.lock and /build/cargo-vendor-dir/Cargo.lock
cairo> Finished cargoSetupPostPatchHook
cairo> updateAutotoolsGnuConfigScriptsPhase
cairo> configuring
cairo> building
cairo> Executing cargoBuildHook
cairo> ++ env CC_X86_64_UNKNOWN_LINUX_GNU=/nix/store/90h6k8ylkgn81k10190v5c9ldyjpzgl9-gcc-wrapper-12.3.0/bin/cc CXX_X86_64_UNKNOWN_LINUX_GNU=/nix/store/90h6k8ylkgn81k10190v5c9ldyjpzgl9-gcc-wrapper-12.3.0/bin/c++ CARGO_TARGET_X86_64_UNKNOWN_LINUX_GNU_LINKER=/nix/store/90h6k8ylkgn81k10190v5c9ldyjpzgl9-gcc-wrapper-12.3.0/bin/cc CC_X86_64_UNKNOWN_LINUX_GNU=/nix/store/90h6k8ylkgn81k10190v5c9ldyjpzgl9-gcc-wrapper-12.3.0/bin/cc CXX_X86_64_UNKNOWN_LINUX_GNU=/nix/store/90h6k8ylkgn81k10190v5c9ldyjpzgl9-gcc-wrapper-12.3.0/bin/c++ CARGO_TARGET_X86_64_UNKNOWN_LINUX_GNU_LINKER=/nix/store/90h6k8ylkgn81k10190v5c9ldyjpzgl9-gcc-wrapper-12.3.0/bin/cc CARGO_BUILD_TARGET=x86_64-unknown-linux-gnu HOST_CC=/nix/store/90h6k8ylkgn81k10190v5c9ldyjpzgl9-gcc-wrapper-12.3.0/bin/cc HOST_CXX=/nix/store/90h6k8ylkgn81k10190v5c9ldyjpzgl9-gcc-wrapper-12.3.0/bin/c++ cargo build -j 64 --target x86_64-unknown-linux-gnu --frozen --profile release
cairo> error: the lock file /build/source/Cargo.lock needs to be updated but --frozen was passed to prevent this
cairo> If you want to try to generate the lock file without accessing the network, remove the --frozen flag and use --offline instead.
```

**Other information:**
https://github.com/NixOS/nixpkgs/pull/273023

**Comment by orizi** [comment]
Doing patch fix 2.4.1 - this issue won't be in it.

**Comment by RaitoBezarius** [comment]
Awesome, thank you!
